src/extensions/extensionsApi.ts

```diff
@@ -149,12 +149,14 @@ export interface Param {
   validationErrorMessage?: string;
   immutable?: boolean;
   example?: string;
+  reconfiguring?: boolean;
```

Comment:
I don't love this data model - we are reconfiguring the extension, not the paramSpec, so this feels wrong. Additionally, we only check it for secret params, but it gets set on all params.

Reply:
Agreed, it was bothering me as well. Made it a param that we pass alone to various "ask param" calls.